### PR TITLE
falcon: add autopatching

### PR DESCRIPTION
**Pull request by talwai**
Add an autopatcher into the falcon integration. The autopatcher will
_not_ run by default under `patch_all()` but will run by default under
`ddtrace-run`

Also replicates the existing falcon test case to run without explicit
patching, instead under `ddtrace-run`

### CODE REVIEW

File: ddtrace/contrib/falcon/middleware.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            status = '404'
+
+        # If we have an active unhandled error, treat it as a 500
+        span.set_traceback()
+        err_msg = span.get_tag(errx.ERROR_MSG)
+        if err_msg and not _is_404(err_msg):
+            status = '500'
+
+        span.set_tag(httpx.STATUS_CODE, status)
+        span.finish()
```
Comment (palazzem) [inline ddtrace/contrib/falcon/middleware.py]: I know that this is not part of the PR, but because we already saw that problem in another place, I think it's better to wrap everything in a `Span` context manager. "Just in case".

Comment (talwai) [inline ddtrace/contrib/falcon/middleware.py]: The behavior we want here is subtly different from the `Span` context manager though. we need to rely on pulling the current span and not creating a new one. we could write a new context manager that has something like : 
```
def __exit__(...):
   self.tracer.current_span().finish()
```

but i think that's a bit out of scope of the PR :D

Comment (palazzem) [inline ddtrace/contrib/falcon/middleware.py]: oh what I meant is just:
```python
span = self.tracer.current_span()
if not span:
    return  # unexpected

with span:
    status = httpx.normalize_status_code(resp.status)
    # the rest of the code without the span.finish()
```

But yeah it's fine no worries :D
